### PR TITLE
Enable named op conversion to xsmm for matmul and brgemm

### PR DESCRIPTION
**Pull request by chelini**
Use xsmm gemm instead of brgemm when batch is 1.

### CODE REVIEW

File: lib/TPP/ConvertLinalgToXsmm.cpp [hunk 1]
```diff
@@ -437,30 +437,47 @@ static void replaceOpWithBrgemm(RewriterBase &rewriter,
   int64_t strideA = brgemmInfo.strideA;
   int64_t strideB = brgemmInfo.strideB;
 
-  DenseI64ArrayAttr dims = DenseI64ArrayAttr::get(
-      rewriter.getContext(),
-      ArrayRef<int64_t>{loops[m], loops[n], loops[k], lda, ldb, ldc, strideA,
-                        strideB});
+  bool hasBatch = batch != std::numeric_limits<unsigned>::max();
```
Comment (rengolin) [inline lib/TPP/ConvertLinalgToXsmm.cpp:440]:
```suggestion
  bool hasBatch = (batch != std::numeric_limits<unsigned>::max());
```

File: lib/TPP/ConvertLinalgToXsmm.cpp [hunk 2]
```diff
@@ -437,30 +437,47 @@ static void replaceOpWithBrgemm(RewriterBase &rewriter,
...
-  Value batchDim = rewriter.create<arith::ConstantOp>(
-      loc, integer64, rewriter.getIntegerAttr(integer64, batchVal));
-  SmallVector<Value> invokeOperands;
-  invokeOperands.push_back(dispatched);
-  invokeOperands.append(linalgOp->getOperands().begin(),
-                        linalgOp->getOperands().end());
-  invokeOperands.push_back(batchDim);
-  rewriter.replaceOpWithNewOp<xsmm::BrgemmOp>(linalgOp, dtype, invokeOperands);
+
+  if (hasBatch) {
```
Comment (rengolin) [inline lib/TPP/ConvertLinalgToXsmm.cpp:447]: These two codes can be merged into one by selecting the flags' strides and adding the `batchVal` inside an `if (hasBatch)` and leaving the rest identical.

Comment (chelini) [reply lib/TPP/ConvertLinalgToXsmm.cpp:450]: I moved the flags out. Note that the dispatch calls are not the same, and in the brgemm we have an extra parameter to pass, the batch.

File: lib/TPP/ConvertLinalgToXsmm.cpp [hunk 3]
```diff
@@ -798,12 +811,46 @@ void ConvertLinalgToXsmm::runOnOperation() {
...
+  using OpRewritePattern<linalg::MatmulOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::MatmulOp matmulOp,
+                                PatternRewriter &rewriter) const override {
+    auto gemmInfo = isMappableToBrgemm(matmulOp);
+    if (failed(gemmInfo)) {
+      llvm::errs() << "Failed\n";
+      return failure();
+    }
+    replaceOpWithBrgemm(rewriter, matmulOp, *gemmInfo);
```
Comment (adam-smnk) [inline lib/TPP/ConvertLinalgToXsmm.cpp:826]: nit: this function name should be updated as it's not only brgemm anymore

**Review by adam-smnk — APPROVED** [review]

**Pull request merged**